src/components/shared/StackImplementationSetting.js

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import { Localized } from '@fluent/react';
+
+import { changeImplementationFilter } from 'firefox-profiler/actions/profile-view';
+import { getImplementationFilter } from 'firefox-profiler/selectors/url-state';
+
+import { toValidImplementationFilter } from 'firefox-profiler/profile-logic/profile-data';
+import explicitConnect, {
+  type ConnectedProps,
+} from 'firefox-profiler/utils/connect';
+
+import { getProfileUsesMultipleStackTypes } from 'firefox-profiler/selectors/profile';
+
+import './StackSettings.css';
+
+import type { ImplementationFilter } from 'firefox-profiler/types';
+
+type OwnProps = {||};
+
+type StateProps = {|
+  +implementationFilter: ImplementationFilter,
+  +allowSwitchingStackType: boolean,
+|};
+
+type DispatchProps = {|
+  +changeImplementationFilter: typeof changeImplementationFilter,
+|};
+
+type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
+
+class StackImplementationSettingImpl extends PureComponent<Props> {
+  _onImplementationFilterChange = (e: SyntheticEvent<HTMLInputElement>) => {
+    this.props.changeImplementationFilter(
+      // This function is here to satisfy Flow that we are getting a valid
+      // implementation filter.
+      toValidImplementationFilter(e.currentTarget.value)
+    );
+  };
+
+  _renderImplementationRadioButton(
+    labelL10nId: string,
+    implementationFilter: ImplementationFilter
+  ) {
+    return (
+      <label className="photon-label photon-label-micro stackSettingsFilterLabel">
+        <input
+          type="radio"
+          className="photon-radio photon-radio-micro"
```

Comment:
I removed the class `stackSettingsFilterInput` because it wasn't specified in CSS anywhere.